### ISSUE TITLE
.NET Core 3.0 apps failed to run with static files in folders published as a single file

### ISSUE THREAD

**Issue by xiaoyuvax** [dotnet/core-setup#8588]
### Describe the bug
Static files included in project but in FOLDER structure(with at least one underlying subfolder) failed to run after published as a single file, eg. you Included a wwwroot folder with a css subfolder in your project and set all individual files to be copied to output folders under VS which are assumed to be packed into the single executable.

running the output executable shows following messages:
> Failure processing application bundle; possible file corruption.
> Invalid FileEntry detected.
> A fatal error was encountered. Could not extract contents of the bundle

These messages were gone when abovementioned all subfolders (such as /wwwroot/css) were removed.
Static files in project root, or in first level folder don't yield this problem.

### To Reproduce
1.Inlucde folders with subfolders of files in project (such as "/wwwroot/css/”)
2.Publish with PublishSingleFile=true 
3.Run the output executable,you get the message.

### Expected behavior
Launch as usual since the packed folders structure should be extracted to the temporary path properly.

### Relevant Issue:
[https://github.com/aspnet/AspNetCore-Tooling/pull/820](url)
[https://github.com/aspnet/AspNetCore/issues/12020](url)

@DamianEdwards @davidfowl @glennc @anurse

**Comment by swaroop-sridhar** [comment]
Transfer to core-setup repo

**Comment by swaroop-sridhar** [comment]
Hi @xiaoyuvax, I just tried reproing the bug with the following steps:
* dotnet new console
* Copy the www folder published in a `dotnet new mvc` project to the console project
* Use the following csproj file for the console project.
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <PublishSingleFile>true</PublishSingleFile>
    <RuntimeIdentifier>win10-x64</RuntimeIdentifier>
    <ApplicationIcon />
    <StartupObject />
  </PropertyGroup>

  <ItemGroup>
    <Content Include=" wwwroot\**\*">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
  </ItemGroup>
</Project>
```

However, I did not repro the failure. I tried a few things like including only portions of the `wwwroot`  content, adding the `<None remove="..."/>` items that you had, etc. But still the app ran OK.

Can you please attach your repro, so that I can run the same app? Thanks.

**Comment by xiaoyuvax** [comment]
Aha,I finally found the cause!

**ZERO LENGTH file in the folder structure may cause the executable bundle UNABLE to extract itself.** 

I guess it is possible a bug on calculating total length of the contents.
It is irrelevant to subfolders at all, i happen to have a zero length file in the subfolder. LOL

**Comment by swaroop-sridhar** [comment]
Thanks for investigating further @xiaoyuvax. This issue is fixed for 3.1 in 
https://github.com/dotnet/core-setup/commit/bbf5542781136f9f3a1f30b010cb782e775d54c7

Dup of https://github.com/dotnet/core-setup/issues/8422
